code/game/objects/effects/temporary_visuals/temporary_visual.dm

```diff
@@ -4,6 +4,8 @@
 	anchored = TRUE
 	layer = ABOVE_MOB_LAYER
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	///When set to true, we manually set the effect's duration via the set_duration proc
+	var/manual_duration = FALSE
```

Comment:
why

Reply:
In this case for hive alert pings; ping time can be modified as an input

Reply:
It should use initialize args instead